modules/apps/27-interchain-accounts/controller/keeper/keeper.go

```diff
@@ -0,0 +1,222 @@
+package keeper
+
+import (
+	"fmt"
+	"strings"
+
+	baseapp "github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	capabilitykeeper "github.com/cosmos/cosmos-sdk/x/capability/keeper"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	"github.com/tendermint/tendermint/libs/log"
+
+	controllertypes "github.com/cosmos/ibc-go/v2/modules/apps/27-interchain-accounts/controller/types"
+	"github.com/cosmos/ibc-go/v2/modules/apps/27-interchain-accounts/types"
+	channeltypes "github.com/cosmos/ibc-go/v2/modules/core/04-channel/types"
+	host "github.com/cosmos/ibc-go/v2/modules/core/24-host"
+)
+
+// Keeper defines the IBC interchain accounts controller keeper
+type Keeper struct {
+	storeKey sdk.StoreKey
+	cdc      codec.BinaryCodec
+
+	ics4Wrapper   types.ICS4Wrapper
+	channelKeeper types.ChannelKeeper
+	portKeeper    types.PortKeeper
+	accountKeeper types.AccountKeeper
+
+	scopedKeeper capabilitykeeper.ScopedKeeper
+
+	msgRouter *baseapp.MsgServiceRouter
+}
+
+// NewKeeper creates a new interchain accounts controller Keeper instance
+func NewKeeper(
+	cdc codec.BinaryCodec, key sdk.StoreKey,
+	ics4Wrapper types.ICS4Wrapper, channelKeeper types.ChannelKeeper, portKeeper types.PortKeeper,
+	accountKeeper types.AccountKeeper, scopedKeeper capabilitykeeper.ScopedKeeper, msgRouter *baseapp.MsgServiceRouter,
+) Keeper {
+
+	// ensure ibc interchain accounts module account is set
+	if addr := accountKeeper.GetModuleAddress(controllertypes.ModuleName); addr == nil {
+		panic("the Interchain Accounts controller module account has not been set")
+	}
+
+	return Keeper{
+		storeKey:      key,
+		cdc:           cdc,
+		ics4Wrapper:   ics4Wrapper,
+		channelKeeper: channelKeeper,
+		portKeeper:    portKeeper,
+		accountKeeper: accountKeeper,
+		scopedKeeper:  scopedKeeper,
+		msgRouter:     msgRouter,
+	}
+}
+
+// Logger returns the application logger, scoped to the associated module
+func (k Keeper) Logger(ctx sdk.Context) log.Logger {
+	return ctx.Logger().With("module", fmt.Sprintf("x/%s-%s", host.ModuleName, controllertypes.ModuleName))
+}
+
+// InitInterchainAccount is the entry point to registering an interchain account.
+// It generates a new port identifier using the owner address, connection identifier,
+// and counterparty connection identifier. It will bind to the port identifier and
+// call 04-channel 'ChanOpenInit'. An error is returned if the port identifier is
+// already in use. Gaining access to interchain accounts whose channels have closed
+// cannot be done with this function. A regular MsgChanOpenInit must be used.
+func (k Keeper) InitInterchainAccount(ctx sdk.Context, connectionID, counterpartyConnectionID, owner string) error {
```

Comment:
The previous account.go file contained two functions:
- `InitInterchainAccount`
- `RegisterInterchainAccount`

As `InitInterchainAccount` belongs to the `controller` submodule and `RegisterInterchainAccount` to the `host`, I decided to move each to their respective keeper.go files. In my opinion this is a little cleaner, but I don't feel particularly strongly about it.